### PR TITLE
ICMSLST-979 Add spreadsheet download for export applications.

### PR DESCRIPTION
**Pull request by MattHolmes123**

### CODE REVIEW

**Review by mikeylockwood — APPROVED** [review]

File: web/tests/utils/test_search.py [hunk 1]
```diff
@@ -727,6 +727,58 @@ def test_get_search_results_spreadsheet(test_data: FixtureData):
...
+        "Exporter",
+        "Agent",
+        "Application Contact",
+    ]
+
+    case_refs = []
+
+    # Iterate over the remaining data rows
+    for row in spreadsheet_rows:
+        row_data = list(row)
```
Comment (kevinetienne) [inline web/tests/utils/test_search.py:776]: what is row at this stage?

File: web/utils/search.py [hunk 2]
```diff
@@ -811,6 +841,30 @@ def _get_spreadsheet_rows(
         )
 
 
+def _get_export_spreadsheet_rows(records: list[ExportResultRow]) -> Iterable[ExportSpreadsheetRow]:
+    for row in records:
+        certificates = ", ".join(row.certificates)
```
Comment (kevinetienne) [inline web/utils/search.py:846]: Should we remove the `None` from `row.certificates`?

**Review by kevinetienne — APPROVED** [review]

File: web/static/web/js/api/download-spreadsheet.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      e.preventDefault();
+      const warning = document.querySelector(warningId);
+
+      handleResultsSpreadsheetDownload(downloadForm.action, searchFormId)
+        .then(() => {
+          warning.style.display = "none";
+        })
+        .catch(reason => {
+          warning.style.display = "block";
+          console.log(reason);
```
[the remark this answers is not in the document]
Comment (MattHolmes123) [reply web/static/web/js/api/download-spreadsheet.js:16]: I can add warn sure.
The user gets the error message displayed in an error banner on the page anyway

File: web/utils/search.py (hunk 2 above)
Comment (MattHolmes123) [reply web/utils/search.py:846]: Im not sure what you mean?

File: web/tests/utils/test_search.py (hunk 1 above)
Comment (MattHolmes123) [reply web/tests/utils/test_search.py:776]: Its a tuple, I don't need to cast to a list :+1:

File: web/utils/search.py (hunk 2 above)
Comment (MattHolmes123) [reply web/utils/search.py:849]: As discussed this is fine.

**Pull request merged**